### PR TITLE
Add warning related to ad and concurrent segment search

### PR DESCRIPTION
**Pull request by jed326**
### Description
Add warning related to ad and concurrent segment search

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/concurrent-segment-search.md [hunk 1]
```diff
@@ -87,6 +87,8 @@ Typically, queries are used with smaller `terminate_after` values and thus compl
 
 Parent aggregations on [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) fields do not support the concurrent search model. Thus, if a search request contains a parent aggregation, the aggregation will be executed using the non-concurrent path even if concurrent segment search is enabled at the cluster level.
 
+The [anomaly-detection](https://opensearch.org/docs/latest/observing-your-data/ad/index/) plugin is not compatible with concurrent segment search. If you wish to use concurrent segment search with the anomaly-detection plugin please apply the index setting to disable concurrent segment search for the anomaly-detection result indices. By default this is pointed to by the alias `.opendistro-anomaly-results`. See [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/12331) for more details.
```
Comment (kolchfa-aws) [inline _search-plugins/concurrent-segment-search.md:90]:
```suggestion
The [Anomaly Detection](https://opensearch.org/docs/latest/observing-your-data/ad/index/) plugin is not compatible with concurrent segment search. To use concurrent segment search with the Anomaly Detection plugin, disable concurrent segment search for the anomaly detection result indexes by applying the index-level setting. By default, the result indexes are specified by the `.opendistro-anomaly-results` alias. See [this GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/12331) for more details.
```

**Review by kolchfa-aws — APPROVED** [review]
Thanks, @jed326! Minor rewrites.

**Comment by sohami** [comment]
@kaituo To review the AD related information

**Review by amitgalitz — APPROVED** [review]
This looks good to me, thank you for adding this

File: _search-plugins/concurrent-segment-search.md [hunk 2]
```diff
@@ -87,6 +87,10 @@ Typically, queries are used with smaller `terminate_after` values and thus compl
 
 Parent aggregations on [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) fields do not support the concurrent search model. Thus, if a search request contains a parent aggregation, the aggregation will be executed using the non-concurrent path even if concurrent segment search is enabled at the cluster level.
 
+The [Anomaly Detection](https://opensearch.org/docs/latest/observing-your-data/ad/index/) plugin is not compatible with concurrent segment search. To use concurrent segment search with the Anomaly Detection plugin, disable concurrent segment search for the anomaly detection result indexes by applying the index-level setting. By default, the result indexes are specified by the `.opendistro-anomaly-results` alias. See [this GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/12331) for more details.
```
Comment (natebower) [inline _search-plugins/concurrent-segment-search.md:90]:
```suggestion
The [Anomaly Detection](https://opensearch.org/docs/latest/observing-your-data/ad/index/) plugin is not compatible with concurrent segment search. To use concurrent segment search with the Anomaly Detection plugin, disable concurrent segment search for the anomaly detection result indexes by applying the index-level setting. By default, the result indexes are specified by the `.opendistro-anomaly-results` alias. See [this GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/12331) for more information.
```

**Review by natebower — COMMENTED** [review]
@jed326 @kolchfa-aws Just one word swap. Thanks!

**Pull request merged**